src/DotLiquid/Tags/For.cs

```diff
@@ -166,7 +166,7 @@ public override void Render(Context context, TextWriter result)
             });
         }
 
-        private static List<object> SliceCollectionUsingEach(Context context, IEnumerable collection, int from, int? to)
+        private static List<object> SliceCollectionUsingEach(Context context, IEnumerable collection, int from, long? to)

```

Comment:
Iteration upper limit needed 64 bit range would exceed by nature the configured iteration limit defined as 32 bit int and make little sense. While collection, say an array, can extend to long index, an iteration action in the 64 bit range of steps would take a prohibitive execution time.